### DISCUSSION TITLE
How to render text with RGB color in statusline?

### DISCUSSION THREAD

**Discussion by linrongbin16** [MunifTanjim/nougat.nvim#78] [Q&A] [answered]
Hi, this is the following question of the issue: https://github.com/MunifTanjim/nougat.nvim/issues/77, which could be off-topic of this plugin...

Now I'm trying to implement the `Item` component for 'filetype':

```lua
-- file type
stl:add_item(Item({
    hl = { bg = color.bg1 },
    sep_left = sep.left_lower_triangle_solid(true),
    prefix = " ",
    suffix = " ",
    type = "lua_expr",
    content = function(ctx)
        message.info("|nougat.filetype| ctx:%s", vim.inspect(ctx))
        local ft =
            vim.api.nvim_get_option_value("filetype", { buf = ctx.bufnr })
        local ok, devicons = pcall(require, "nvim-web-devicons")
        if not ok then
            return ft or ""
        end
        local icon_text, icon_color = devicons.get_icon_color_by_filetype(ft)
        message.info(
            "|nougat.filetype| ctx:%s, icon_text:%s, icon_color:%s",
            vim.inspect(ctx),
            vim.inspect(icon_text),
            vim.inspect(icon_color)
        )
        if icon_text and icon_color then
            return tostring(icon_text) .. " " .. ft
        else
            return ft or ""
        end
    end,
}))
```

I don't know how to render the RGB color code to statusline:

<img width="1716" alt="image" src="https://github.com/MunifTanjim/nougat.nvim/assets/6496887/ffafdbb4-4aab-448a-b51b-84e18e2b7b0f">

If the color of the icon text is rendered, it should looks like:

<img width="396" alt="image" src="https://github.com/MunifTanjim/nougat.nvim/assets/6496887/40f3865b-12c6-4744-9e6b-e06f8a96e6c0">

I googled some questions, but they all told me to define a syntax highlighting group first... 

I'm not sure is it already the best solution?

**Comment by MunifTanjim** [comment, marked as answer]
> I googled some questions, but they all told me to define a syntax highlighting group first...
>
> I'm not sure is it already the best solution?

Nougat has a utility function for this:

```lua
local get_hl_name = require("nougat.color").get_hl_name
```

You can use it like this:

```lua
stl:add_item({
  hl = { bg = "yellow" },
  content = function(item, ctx)
    return core.highlight(get_hl_name({ fg = "red" }, item.hl or ctx.hl)) .. "CONTENT"
  end
})
```

Here, `{ fg = "red" }` is the highlight definition you want. `item.hl or ctx.hl` is the fallback highlight definition for the current context.

---

Also for filetype icon, I just added `nut.buf.filetype_icon`: https://github.com/MunifTanjim/nougat.nvim/pull/79

You can use it like this:

```lua
stl:add_item({
  hl = { bg = color.bg1 },
  prefix = " ",
  content = {
    nut.buf.filetype_icon({ suffix = " " }),
    nut.buf.filetype({}),
  },
  suffix = " ",
})
```

**Reply by linrongbin16** [reply]
thanks! it works great!
